Modules/_io/_iomodule.c

```diff
@@ -368,6 +367,7 @@ _io_open_impl(PyObject *module, PyObject *file, const char *mode,
         if (blksize_obj == NULL)
             goto error;
         buffering = PyLong_AsLong(blksize_obj);
+        buffering = Py_MAX(Py_MIN(buffering, 8192 * 1024), DEFAULT_BUFFER_SIZE);
```

Comment:
This needs to go below the error check, otherwise the -1 return value PyErr could be ignored.